meta.go

```diff
@@ -241,16 +242,90 @@ func parseStringList(list interface{}) []string {
 	return result
 }
 
-var termNameRE = regexp.MustCompile("^[a-z]+([a-z0-9-]+)(/[0-9]+)?$")
+var validTermName = regexp.MustCompile(`^[a-zA-Z][\w-]+$`)
 
-func checkTerm(s string) error {
-	match := termNameRE.FindStringSubmatch(s)
-	if match == nil {
-		return fmt.Errorf("invalid term name %q: must match %s", s, termNameRE.String())
+// Term represents a single term URI. The term can either be owned
+// or "public" (meaning there is no owner).
+type Term struct {
+	Tenant   string
+	Owner    string
+	Name     string
+	Revision int
```

Comment:
Perhaps document these fields?
In particular, it might be worth documenting that revisions start from 1 (unlike charm ids) so that it's clear that a zero Revision implies that the revision has not been set in the terms id.